scopesim/optics/fov.py

```diff
@@ -979,29 +987,21 @@ def spline_order(self) -> int:
         return from_currsys("!SIM.computing.spline_order", self.cmds)
 
     @property
-    def cube_fields(self):
-        """Return list of non-BG_SRC ImageHDU fields with NAXIS=3."""
-        return [field.field for field in self.fields
-                if isinstance(field, CubeSourceField)]
-
-    @property
-    def image_fields(self):
-        """Return list of non-BG_SRC ImageHDU fields with NAXIS=2."""
-        return [field.field for field in self.fields
-                if isinstance(field, ImageSourceField)]
-
-    @property
-    def table_fields(self):
-        """Return list of Table fields."""
-        return [field.field for field in self.fields
-                if isinstance(field, TableSourceField)]
+    def fields(self) -> list[SourceField]:
+        """Return list of all fields, for backwards compatibility."""
+        return (self.cube_fields + self.image_fields + self.table_fields +
+                self.background_fields)
 
     @property
-    def background_fields(self):
-        """Return list of BG_SRC ImageHDU fields."""
-        # HACK: While Sourcefields are not fully supported.
-        return [fits.ImageHDU(header=field.header) for field in self.fields
-                if isinstance(field, BackgroundSourceField)]
+    def spectra(self) -> dict[int, SourceSpectrum]:
+        """Return a collection of all fields' spectra."""
+        specs = {}
+        for field in self.fields:
+            if not isinstance(field, SpectrumSourceField):
+                continue
+            for ref, spec in field.spectra.items():
+                specs[ref] = spec
```

Comment:
```suggestion
    def spectra(self) -> dict[int, SourceSpectrum]:
        """Return a collection of all fields' spectra."""
        return {
            ref: spec
            for field in [f for f in self.fields if isinstance(f, SpectrumSourceField)]
            for ref, spec in field.spectra.items()
        }
```
Not sure whether this is better

Reply:
Sure, don't know why I didn't think of the dict comprehension here...